spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -1888,6 +1888,88 @@ public void testZOrderRewriteWithSpecificOutputSpecId() {
     shouldRewriteDataFilesWithPartitionSpec(table, outputSpecId);
   }
 
+  @TestTemplate
+  public void testRewriteMaxFilesOption() {
```

Comment:
We need this test in core. As the feature is moved to core

Reply:
Great point ! I did move all the tests from spark module to `TestBinPackRewriteFilePlanner` class